### PR TITLE
Blockchain Connector: Add functional tests - Closes #1022

### LINKED ISSUE #1022: Blockchain Connector: Add functional tests
### Description

Create functional tests for the following features:
 
- Genesis block import
- Helper functions in blockchain-coordinator: schema-based decoder/encoder
- ...

### Acceptance Criteria

- Tests are written and running with Jest

### PR DESCRIPTION
**Pull request by nagdahimanshu**
### What was the problem?
This PR resolves #1022 

### How was it solved?
- [x] Add functional tests
  - [x] Genesis block import
  - [x] Validate genesis block
  - [x] Encoder
  - [x] Decoder
- [x] Add logic to run connector in testing mode and expose methods required only for testing
- [x] Add logic to download genesis block SHA256 file
- [x] Remove 'core' microservice

### How was it tested?
Local

### CODE REVIEW

File: services/blockchain-connector/tests/functional/decoder.test.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+			},
+		);
+		expect(result).toMatchObject(block);
+	});
+
+	it('decode response', async () => {
+		const result = await broker.call('connector.decodeResponse',
+			{
+				action: 'app_getBlockByHeight',
+				response: encodedBlock,
```
Comment (sameersubudhi) [inline services/blockchain-connector/tests/functional/decoder.test.js:92]: Please refactor `action` to `endpoint`, as requested above.

**Review by sameersubudhi — CHANGES_REQUESTED** [review]
Please consider the following remarks.

File: services/blockchain-connector/tests/functional/genesisBlock.test.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+describe('Genesis Block import tests', () => {
+	let genesisBlockFilePath;
+
+	beforeAll(() => broker.start());
+	afterAll(() => broker.stop());
+
+	it('Verify if genesis block is downloaded successfully', async () => {
+		const nodeInfo = await broker.call('connector.getNetworkStatus');
+		genesisBlockFilePath = `./data/${nodeInfo.data.networkIdentifier}/genesis_block.json.tar.gz`;
```
[the remark this answers is not in the document]
Comment (nagdahimanshu) [reply services/blockchain-connector/tests/functional/genesisBlock.test.js:37]: Discussed offline N/A.

File: services/blockchain-connector/methods/tests/decoder.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+	},
+	{
+		name: 'decodeTransactionSerialized',
+		params: {
+			encodedTransaction: { optional: false, type: 'string' },
+		},
+		controller: ({ encodedTransaction }) => {
+			const decodedTransaction = decodeTransaction(encodedTransaction);
+			return parseToJSONCompatObj(decodedTransaction);
+		},
```
Comment (sameersubudhi) [inline services/blockchain-connector/methods/tests/decoder.js:41]: Please keep it consistent with the remaining.

File: services/blockchain-connector/tests/functional/decoder.test.js [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+	it('decode event payload', async () => {
+		const result = await broker.call('connector.decodeEventPayload',
+			{
+				eventName: 'app_newBlock',
+				payload: {
+					block: encodedBlock,
+				},
+			},
+		);
```
Comment (sameersubudhi) [inline services/blockchain-connector/tests/functional/decoder.test.js:89]:
```suggestion
		const result = await broker.call('connector.decodeEventPayload', {
			eventName: 'app_newBlock',
			payload: { block: encodedBlock },
		});
```

File: services/blockchain-connector/tests/functional/decoder.test.js [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+	});
+
+	it('decode block with transaction', async () => {
+		const result = await broker.call('connector.decodeBlockSerialized', { encodedBlock: encodedBlockWithTransaction });
+		expect(result).toMatchObject({
+			header: expect.any(Object),
+			assets: expect.any(Object),
+			transactions: expect.any(Object),
+		});
+		blockWithTransaction.transactions[0].size = result.transactions[0].size;
```
Comment (sameersubudhi) [inline services/blockchain-connector/tests/functional/decoder.test.js:58]: Please remove this line and ensure the constants already have the expected values.

File: services/blockchain-connector/tests/functional/decoder.test.js [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+		expect(result).toMatchObject(block);
+	});
+
+	it('decode response', async () => {
+		const result = await broker.call('connector.decodeResponse',
+			{
+				endpoint: 'app_getBlockByHeight',
+				response: encodedBlock,
+			},
+		);
```
Comment (sameersubudhi) [inline services/blockchain-connector/tests/functional/decoder.test.js:99]:
```suggestion
		const result = await broker.call('connector.decodeResponse', {
			endpoint: 'app_getBlockByHeight',
			response: encodedBlock,
		});
```

File: services/blockchain-connector/tests/functional/genesisBlock.test.js [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+		const dataHash = crypto.createHash('sha256');
+
+		const fileHash = await new Promise(resolve => {
+			fileStream.on('data', (data) => {
+				dataHash.update(data);
+			});
+
+			fileStream.on('end', () => {
+				resolve(dataHash.digest());
+			});
```
Comment (sameersubudhi) [inline services/blockchain-connector/tests/functional/genesisBlock.test.js:61]:
```suggestion
			fileStream.on('data', (data) => { dataHash.update(data); });
			fileStream.on('end', () => { resolve(dataHash.digest()); });
```

**Review by sameersubudhi — CHANGES_REQUESTED** [review]

**Review by sameersubudhi — APPROVED** [review]
LGTM 👍

**Pull request merged**